### ISSUE TITLE
pdo_sqlsrv and sqlsrv should be differentiated.

### ISSUE THREAD

**Issue by Arkasian** [MicrosoftDocs/sql-docs#5461]
pdo_sqlsrv and sqlsrv are different extensions. There is no need for anyone to install them both.
For example, when using Symfony and Doctrine 2, I would need just sqlsrv, pdo_sqlsrv is useless cause it causes Doctrine to fail as it is stated [here](https://www.doctrine-project.org/projects/doctrine-dbal/en/2.8/reference/configuration.html#pdo-sqlsrv-sqlsrv).
What I mean, is too mention that those libs are parallel to each other and warn user to be aware that pdo_sqlsrv or sqlsrv can cause similar issues to that one I reffered to above. What's more, why to install them both if both of them are somehow fully functional?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 35f591be-6564-c959-7a92-6489bdd4f60b
* Version Independent ID: eec72536-eda0-1bc8-a41a-36ed4b9a8883
* Content: [Linux and macOS Installation for the Drivers for PHP - SQL Server](https://docs.microsoft.com/en-us/sql/connect/php/installation-tutorial-linux-mac?view=sql-server-ver15)
* Content Source: [docs/connect/php/installation-tutorial-linux-mac.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/connect/php/installation-tutorial-linux-mac.md)
* Product: **sql**
* Technology: **connectivity**
* GitHub Login: @David-Engel
* Microsoft Alias: **v-daenge**

**Comment by WilliamAntonRohm** [comment]
@Arkasian -- Lukas, thank you for your feedback.

@David-Engel -- David, please look into this issue.

**Comment by David-Engel** [comment]
@Arkasian 

> For example, when using Symfony and Doctrine 2, I would need just sqlsrv, pdo_sqlsrv is useless cause it causes Doctrine to fail as it is stated [here](https://www.doctrine-project.org/projects/doctrine-dbal/en/2.8/reference/configuration.html#pdo-sqlsrv-sqlsrv).

The only problem I see noted at that link is this comment "Note that this driver caused problems in our tests. Prefer the sqlsrv driver if possible." But I am not aware of any problems in the latest release that are caused by them being both installed and there are no details provided.

> What I mean, is too mention that those libs are parallel to each other and warn user to be aware that pdo_sqlsrv or sqlsrv can cause similar issues to that one I reffered to above.

They should not cause issues when installed together. Any issues can and should be filed in the [GitHub repo](https://github.com/microsoft/msphpsql/issues) for the driver so that they can be appropriately addressed.

> What's more, why to install them both if both of them are somehow fully functional?

You are free to simply install one or the other if you know what you need. The pdo_sqlsrv driver adheres to the PDO API specification and may not be able to offer SQL Server-specific functionality that is available in the sqlsrv driver. However PDO may be sufficient for your needs and/or fit a framework you are using that requires the PDO standard in its database connector.

Regards,
David

**Comment by Arkasian** [comment]
@David-Engel 
Thank you very much for your input. I didn't know it, I have gladly written all this explanation. And what I meant is, a lot of php developer, just like me wouldn't know that too, so thus I asked, from the very beginning of this issue, to just add sort of explanation you jsut give, to documentation as a kind of Notice. I didn't want to say, that anything does not work or anything collides. I wanted to prompt that I, as a developer of php using mainly mysql, pgsql, didn't know it, so I thought it would be fine to leave answers to questions I had in this document on Wiki :) That's all.

**Comment by David-Engel** [comment]
I added a statement near the top of the page explaining that the steps can be customized to just install one or the other. It should appear sometime today.

Thanks,
David

#please-close

**Comment by Arkasian** [comment]
@David-Engel I just noted that there is some difference for Doctrine exactly, I don't know how it is about Eloquent.
But it makes difference if you install one or both. When I had both installed, Doctirne somehow didn't want to use sqlsrv, because when I started connectrion string with "sqlsrv" while having both installed, it threw issues with no matching driver etc. And just a few minutes ago, I tried not installing pdo_sqlsrv and just sqlsrv and it works pretty fine. Consider this message just an update, I will search for issues in Doctrine github, as I believe it is on their side to note it on their documentation and not just mention that there is some issue. Once again, thanks for your help!
